### PR TITLE
dpp: init at 10.0.26

### PR DESCRIPTION
**Pull request by pongo1231**
## Description of changes

C++ Discord API Bot Library

https://dpp.dev/index.html

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lelgenio** [comment]
Result of `nixpkgs-review pr 259251` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dpp</li>
  </ul>
</details>

**Comment by pongo1231** [comment]
Moved derivation to pkgs/by-name/

File: pkgs/by-name/dp/dpp/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+, libopus
+}:
+
+stdenv.mkDerivation rec {
+  pname = "DPP";
+  version = "10.0.26";
+
+  src = fetchFromGitHub {
+    owner = "brainboxdotcc";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/dp/dpp/package.nix:18]:
```suggestion
  pname = "dpp";
  version = "10.0.26";

  src = fetchFromGitHub {
    owner = "brainboxdotcc";
    repo = "DPP";
```
the attr and pname should ideally match

File: pkgs/by-name/dp/dpp/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-o78ijctDqrONyi8A3+FXvnK9q97B4j1ZIQYNUgl6XJU=";
+  };
+
+  nativeBuildInputs = [ cmake pkg-config ];
+
+  buildInputs = [ zlib openssl libsodium libopus ];
+
+  meta = with lib; {
+    description = "C++ Discord API Bot Library - D++ is Lightweight and scalable for small and huge bots!";
```
Comment (SuperSandro2000) [inline pkgs/by-name/dp/dpp/package.nix:28]:
```suggestion
    description = "Lightweight and efficient library for Discord written in modern C++";
```

That's a better description

**Comment by pongo1231** [comment]
Seemingly already added by https://github.com/NixOS/nixpkgs/pull/281320.

**Pull request closed without merging**